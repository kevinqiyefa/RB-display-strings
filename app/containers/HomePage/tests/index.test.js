/**
 *
 * Tests for HomePage
 *
 * @see https://github.com/react-boilerplate/react-boilerplate/tree/master/docs/testing
 *
 */

import React from 'react';
import { render } from 'react-testing-library';

import { IntlProvider } from 'react-intl';
import { Provider } from 'react-redux';
import { browserHistory } from 'react-router-dom';

// import { changeUsername } from '../actions';
// import { loadRepos } from '../../App/actions';
import { HomePage } from '../index';
import configureStore from '../../../configureStore';

describe('<HomePage />', () => {
  let store;

  beforeAll(() => {
    store = configureStore({}, browserHistory);
  });

  it('should render and match the snapshot', () => {
    const fetchData = jest.fn();
    const {
      container: { firstChild },
    } = render(
      <Provider store={store}>
        <IntlProvider locale="en">
          <HomePage
            strings={[{ id: 1, string: 'text' }]}
            fetchStringRequest={fetchData}
          />
        </IntlProvider>
      </Provider>,
    );
    expect(firstChild).toMatchSnapshot();
  });
});
