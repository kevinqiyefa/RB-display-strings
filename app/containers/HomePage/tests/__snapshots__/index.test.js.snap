// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HomePage /> should render and match the snapshot 1`] = `
<div
  class="ContentWrapper-sc-1pl7b9y-0 kNVNXH"
>
  <h1
    class="DisplayStrings__ContentHeader-sc-1743nj8-0 jZIUMu"
  >
    Strings From Database:
  </h1>
  <ul
    class="UL-sc-16unqrb-0 PCfYS"
  >
    <li
      class="DisplayStrings__LI-sc-1743nj8-1 bYXRYq"
    >
      text
    </li>
  </ul>
</div>
`;
